### PR TITLE
[addon] add several for python related documentation parts

### PR DESCRIPTION
**Pull request by AlwinEsch**
Again few parts of my documentation.

This are related partly to python but never been used before on Kodi code before and can be added independent.

### CODE REVIEW

File: xbmc/SortFileItem.h [hunk 1]
```diff
@@ -19,51 +19,61 @@
...
-  SORT_METHOD_LISTENERS,
-  SORT_METHOD_UNSORTED,
-  SORT_METHOD_CHANNEL,
-  SORT_METHOD_CHANNEL_NUMBER,
-  SORT_METHOD_BITRATE,
-  SORT_METHOD_DATE_TAKEN,
-  SORT_METHOD_MAX
+  SORT_METHOD_NONE=0,                             //!< Integer: __0__ - Do not sort
+  SORT_METHOD_LABEL,                              //!< Integer: __1__ - Sort by label
+  SORT_METHOD_LABEL_IGNORE_THE,                   //!< Integer: __2__ - Sort without the label
```
Comment (Razzeee) [inline xbmc/SortFileItem.h]: This might be a little unclear if you don't see the left handed part?

Comment (AlwinEsch) [inline xbmc/SortFileItem.h]: You mean something like `SORT_METHOD_LABEL`? If yes this becomes added from doxygen itself.

File: xbmc/SortFileItem.h [hunk 2]
```diff
@@ -19,51 +19,61 @@
...
+  SORT_METHOD_VIDEO_TITLE,                        //!< Integer: __25__ - Sort by video title
+  SORT_METHOD_VIDEO_SORT_TITLE,                   //!< Integer: __26__ - Sort by video sort title
+  SORT_METHOD_VIDEO_SORT_TITLE_IGNORE_THE,        //!< Integer: __27__ - Sort without video sort title
+  SORT_METHOD_PRODUCTIONCODE,                     //!< Integer: __28__ - Sort by song rating
+  SORT_METHOD_SONG_RATING,                        //!< Integer: __29__ - Sort by song user rating
+  SORT_METHOD_SONG_USER_RATING,                   //!< Integer: __30__ - Sort by user rating
+  SORT_METHOD_MPAA_RATING,                        //!< Integer: __31__ - Sort by MPAA rating
+  SORT_METHOD_VIDEO_RUNTIME,                      //!< Integer: __32__ - Sort by video runtime
+  SORT_METHOD_STUDIO,                             //!< Integer: __33__ - Sort by studio
+  SORT_METHOD_STUDIO_IGNORE_THE,                  //!< Integer: __34__ - Sort without the studio
```
Comment (anaconda) [inline xbmc/SortFileItem.h]: `Sort by studio, ignoring article (f.e. "Universal Pictures" _before_ "The Weinstein Company")`

File: xbmc/SortFileItem.h (hunk 1 above)
Comment (Razzeee) [inline xbmc/SortFileItem.h]: Well if you can see ```SORT_METHOD_LABEL_IGNORE_THE``` it's not that bad as it explains what it does. 
``` Sort without the label``` is confusing better might be ``` Sort by label without 'the'```

**Comment by AlwinEsch** [comment]
Ah now I understand, Thanks, becomes changed to look better.

Here is the current view of them on website:
http://alwinesch.github.io/a00373.html

**Comment by AlwinEsch** [comment]
Is updated and changed. Also added it now as text before value to allow bigger text in future if needed.

**Comment by tamland** [comment]
I think you may have missunderstood the python sorting enums. The python constants are in ModuleXbmcplugin.h and have _nothing_ to do with those. It just happened to be mapped to an underlying enum value and forwarded by `addSortMethod`. But this is an implementation detail and should NOT be leaked into the documentation.

Any other documentation for these values should be put in there, not in SortFileItem.h

**Comment by AlwinEsch** [comment]
I have seen there on function:
`void addSortMethod(int handle, int sortMethod, const String& label2Mask = emptyString);`
the use of integer, but have seen also the e.g. `SWIG_CONSTANT(int,SORT_METHOD_DATE);` where this integers becomes used.

On the binary add-ons becomes this function also be added and has used this as source for both.

Is a integer value not direct usable if not set with `SWIG_CONSTANT`?

**Comment by tamland** [comment]
If you expand the macro you'll see it just creates a local alias, so yeah, only those are accessible by python. Afaik `SORT_METHOD` is also deprecated and only kept for python, so if you plan to add sorting to the binary API you might want to use `SortBy` instead.

**Comment by AlwinEsch** [comment]
Thanks a lot @tamland, I add it then to `ModuleXbmcplugin.h` and for this request I leave away. 

Want to insert the changes on the python interface files like ModuleXbmcplugin.h after it is sure the new documentation part becomes used. Not sure how current doxygen version handle changes on this files, thats why later.

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**